libmesabox/src/lsb/mount/mod.rs

```diff
@@ -0,0 +1,803 @@
+        }
+        Ok(self
+            .path_map
+            .get(dir)
+            .ok_or(MountError::UuidNotFoundError(
+                dir.to_string_lossy().to_string(),
+            ))?.as_path())
+    }
+}
+
+/// Translate Label into corresponding device path
+struct Label {
+    // map Label to actual path
+    path_map: HashMap<OsString, PathBuf>,
+}
+
+impl Label {
+    fn new() -> MountResult<Self> {
+        let mut path_map: HashMap<OsString, PathBuf> = HashMap::new();
+        let dir = fs::read_dir("/dev/disk/by-label").or(Err(MountError::LabelSupportError))?;
+        for symlink in dir {
+            let link = symlink.or(Err(MountError::LabelSupportError))?;
+            path_map.insert(
+                link.file_name(),
+                link.path()
+                    .canonicalize()
+                    .or(Err(MountError::LabelSupportError))?,
+            );
+        }
+        Ok(Self { path_map: path_map })
+    }
+
+    fn get_device_path(&self, input: &OsString) -> MountResult<&Path> {
+        let input_string = input.to_string_lossy();
+        let dir;
+        if input_string.starts_with("Label=") {
+            dir = OsStr::new(&input_string[6..]);
+        } else {
+            dir = input;
+        }
+        Ok(self
+            .path_map
+            .get(dir)
+            .ok_or(MountError::LabelNotFoundError(
+                dir.to_string_lossy().to_string(),
+            ))?.as_path())
+    }
+}
+
+lazy_static! {
+    static ref OPTION_MAP: HashMap<OsString, c_ulong> = {
+        let mut option_map = HashMap::new();
+        option_map.insert(OsString::from("async"), !libc::MS_SYNCHRONOUS);
+        option_map.insert(OsString::from("atime"), !libc::MS_NOATIME);
+        option_map.insert(OsString::from("dev"), !libc::MS_NODEV);
+        option_map.insert(OsString::from("exec"), !libc::MS_NOEXEC);
+        option_map.insert(OsString::from("noatime"), libc::MS_NOATIME);
+        option_map.insert(OsString::from("nodev"), libc::MS_NODEV);
+        option_map.insert(OsString::from("noexec"), libc::MS_NOEXEC);
+        option_map.insert(OsString::from("nosuid"), libc::MS_NOSUID);
+        option_map.insert(OsString::from("remount"), libc::MS_REMOUNT);
+        option_map.insert(OsString::from("ro"), libc::MS_RDONLY);
+        option_map.insert(OsString::from("rw"), !libc::MS_RDONLY);
+        option_map.insert(OsString::from("suid"), !libc::MS_NOSUID);
+        option_map.insert(OsString::from("sync"), libc::MS_SYNCHRONOUS);
+        option_map
+    };
+}
+
+struct Flag {
+    flag: MsFlags,
+}
+
+impl Default for Flag {
+    fn default() -> Self {
+        Self {
+            flag: MsFlags::MS_SILENT,
+        }
+    }
+}
+
+impl Flag {
+    fn from(options: &mut Vec<OsString>) -> MountResult<MsFlags> {
+        //let mut flag = Flag::default();
+        let mut flag = Flag::default().flag.bits();
+        if options.contains(&OsString::from("default")) {
+            options.extend_from_slice(&[
```

Comment:
You shouldn't need to use `OsString` here.  If you use a `Vec<Cow<OsStr>>` instead you can probably just add `Cow::Borrowed(OsStr::new("blah"))` instead of `OsString::from("blah")`.  Another option is to move the `for` loop below into a function and call it for the "default" options and then like how you already do below.

Secondary issue is that if `options` contains `default`, the first line in the `for` loop will cause the function to return an error as `OPTION_MAP` does not have a key called `default`.

Reply:
Thanks, I didn't realize I could use `Cow` here. And I add the `default` option in `OPTION_MAP`